ui/apps/platform/src/Containers/VulnMgmt/List/NodeComponents/VulnMgmtListNodeComponents.js

```diff
@@ -0,0 +1,228 @@
+            Cell: ({ original }) => {
+                const { topVuln } = original;
+                if (!topVuln) {
+                    return (
+                        <div className="mx-auto flex flex-col">
+                            <span>–</span>
+                        </div>
+                    );
+                }
+                const { cvss, scoreVersion } = topVuln;
+                return <TopCvssLabel cvss={cvss} version={scoreVersion} />;
+            },
+            id: componentSortFields.TOP_CVSS,
+            accessor: 'topVuln.cvss',
+            sortField: componentSortFields.TOP_CVSS,
+        },
+        {
+            Header: `Source`,
+            headerClassName: `w-1/8 ${defaultHeaderClassName}`,
+            className: `w-1/8 ${defaultColumnClassName}`,
+            id: componentSortFields.SOURCE,
+            accessor: 'source',
+            sortField: componentSortFields.SOURCE,
+        },
+        {
+            Header: `Location`,
+            headerClassName: `w-1/8 ${defaultHeaderClassName}`,
+            className: `w-1/8 word-break-all ${defaultColumnClassName}`,
+            Cell: ({ original }) => original.location || 'N/A',
+            id: componentSortFields.LOCATION,
+            accessor: 'location',
+            sortField: componentSortFields.LOCATION,
+        },
+        {
+            Header: `Nodes`,
+            entityType: entityTypes.NODE,
+            headerClassName: `w-1/8 ${defaultHeaderClassName}`,
+            className: `w-1/8 ${defaultColumnClassName}`,
+            id: componentSortFields.NODE_COUNT,
+            accessor: 'nodeCount',
+            Cell: ({ original, pdf }) => (
+                <TableCountLink
+                    entityType={entityTypes.NODE}
+                    count={original.nodeCount}
+                    textOnly={pdf}
+                    selectedRowId={original.id}
+                />
+            ),
+            sortField: componentSortFields.NODE_COUNT,
+        },
+        {
+            Header: `Risk Priority`,
+            headerClassName: `w-1/10 ${defaultHeaderClassName}`,
+            className: `w-1/10 ${defaultColumnClassName}`,
+            id: componentSortFields.PRIORITY,
+            accessor: 'priority',
+            sortField: componentSortFields.PRIORITY,
+        },
+    ];
+
+    const componentColumnsBasedOnContext = getFilteredComponentColumns(tableColumns, workflowState);
+
+    return removeEntityContextColumns(componentColumnsBasedOnContext, workflowState);
+}
+
+const VulnMgmtNodeComponents = ({ selectedRowId, search, sort, page, data, totalResults }) => {
+    const query = gql`
+        query getComponents($query: String, $pagination: Pagination) {
+            results: nodeComponents(query: $query, pagination: $pagination) {
+                ...nodeComponentFields
+            }
+            count: nodeComponentCount(query: $query)
+        }
+        ${VULN_NODE_COMPONENT_LIST_FRAGMENT}
+    `;
```

Comment:
I wonder if it is worth doing 2 different sets of files at this "container" level. We should compare the complexity of the single CVE list file with conditionals for the three new types of CVEs, and decide which way is better. Then we made them consistent one way or the other.

Reply:
We can try to leave this as-is for now, and come back tomorrow and make refactors so it's more consistent with your changes